blockchain/blocks/src/header.rs

```diff
@@ -279,6 +279,22 @@ impl BlockHeader {
         self.cached_cid = Cid::from_bytes_default(&self.cached_bytes).map_err(|e| e.to_string())?;
         Ok(())
     }
+    /// Check to ensure block signature is valid
+    pub fn check_block_signature(&self, addr: &Address) -> Result<(), Error> {
```

Comment:
Is there any reason to make the address a param? Aren't we always validating against the miner?

Reply:
You need the address in order to validate the signature

Reply:
`addr == self.miner_address`, doesn't seem like it needs to be passed in

Reply:
Different addresses, when we check_block_signature we are passing in the raw worker miner address not the address included in the header. See https://github.com/filecoin-project/lotus/blob/master/chain/sync.go#L611